### ISSUE TITLE
[TESTING DOC] Add doc for planning customization of input arguments - tensors, kwargs

### ISSUE THREAD

**Issue by anubhavjana** [torch-spyre/torch-spyre#1594]
Currently upstream pytorch test framework generates inputs automatically by default without any explicit control. This works for most cases, but sometimes we need explicit control, for e.g. to generate shapes to handle spyre specific tensor layouts:

- A module test might need a specific tensor shape (e.g. BatchNorm2d might need NCHW input)
- An op test might need a non-contiguous memory layout to exercise a specific code path
- A test takes scalar, None, or slice arguments alongside tensors
- We want to pin inputs to shapes observed during model tracing

Add a planning documentation for the same.
